src/main/kotlin/com/fortysevendeg/arrowinpractice/endpoints/CharacterDetails.kt

```diff
@@ -18,15 +52,36 @@ import io.ktor.routing.get
  */
 fun Routing.characterDetailsEndpoint(charactersDB: CharactersDatabase) {
   authenticate {
-    get("/characters/{characterId}") {
-      val param = call.request.path().substringAfterLast("/")
-      try {
-        val characterId = param.toLong().characterId()
-        val maybeCharacter = charactersDB.getById(characterId)
-        maybeCharacter?.let { character -> call.respond(character) } ?: throw NotFoundException()
-      } catch (e: NumberFormatException) {
-        throw InvalidIdException()
+    getIO("/characters/{characterId}") { app ->
+      monad().binding {
```

Comment:
what If we omit the monad bindings? Are the attendees more familiarized with do notation than with flatMap itself? even if they're familiarized with do notation (probably just the ones with some Haskell background) and being fair, this doesn't really look pretty much like Haskell do notation anyways in terms of syntax. So I feel like it could fit better to teach how to use plain flatMap even if it looks a bit dirtier, or maybe have a gradual understanding on both things step by step, like having it first with flatMap then improving it to use bindings in some way. They for sure understand what's flatMap even if they call it `>>=` or whatever the symbol they are used to use is, but they know about it's nature and how to use it.

thoughts?